packages/app-council/src/Overview/Vote.tsx

```diff
@@ -0,0 +1,283 @@
+
+  protected accountHelp = (): string => this.props.t('This account will be use to approve or disapprove each candidate.');
+
+  protected txMethod = (): string => 'elections.setApprovals';
+
+  protected txParams = (): [boolean[] | null, VoteIndex, BN] => {
+    const { electionsInfo: { voteCount } } = this.props;
+    const { approvals, hint } = this.state;
+
+    return [
+      approvals, new VoteIndex(voteCount), hint
+    ];
+  }
+
+  protected isDisabled = (): boolean => {
+    const { accountId } = this.state;
+
+    return !accountId || this.areVotesUnchanged();
+  }
+
+  protected renderTrigger = (): React.ReactNode => {
+    const { t } = this.props;
+
+    return (
+      <Button
+        isPrimary
+        label={t('Vote')}
+        labelIcon='check'
+        onClick={this.showModal}
+      />
+    );
+  }
+
+  protected renderContent = (): React.ReactNode => {
+    const { electionsInfo: { candidates }, t } = this.props;
+    const { approvals, oldApprovals } = this.state;
+
+    return (
+      <>
+        {
+          oldApprovals && (
+            <article
+              className='warning padded'
+              style={{
+                display: 'flex',
+                alignItems: 'center',
+                margin: '0.5rem 0'
+              }}
+            >
+              <div style={{ flex: '1 1' }}>
+                <Icon name='warning sign' />
+                {t('You have already voted in this round')}
+              </div>
+              <Button.Group style={{ margin: 0 }}>
+                <Button
+                  isDisabled={this.areVotesUnchanged()}
+                  isPrimary
+                  label={t('Reset votes')}
+                  onClick={this.onResetVotes}
+                />
+              </Button.Group>
+            </article>
+          )
+        }
+        <Candidates>
+          {
+            candidates.map((accountId, index): React.ReactNode => {
+              if (!approvals) {
+                return null;
+              }
+              const { [index]: isAye } = approvals;
+              return (
+                <Candidate
+                  className={isAye ? 'aye' : 'nay'}
+                  key={accountId.toString()}
+                  onClick={(): void => this.onChangeVote(index)()}
+                >
+                  <AddressRow
+                    isInline
+                    value={accountId}
+                  >
+                    {this.renderToggle(index)}
+                  </AddressRow>
+                </Candidate>
+              );
+            })
+          }
+        </Candidates>
+      </>
+    );
+  }
+
+  private renderToggle = (index: number): React.ReactNode => {
+    const { t } = this.props;
+    const { approvals } = this.state;
+
+    if (!approvals) {
+      return null;
+    }
+
+    const { [index]: bool } = approvals;
+
+    return (
+      <Toggle
+        label={
+          bool
+            ? (
+              <b style={{ color: 'green' }}>
```

Comment:
Can we extract this one and the red one? (Basically when we have `{ color: 'green' )` as a constant, we skip the creation of the new object each time.